nemo_curator/utils/distributed_utils.py

```diff
@@ -237,6 +237,8 @@ def read_single_partition(
         read_kwargs = {"lines": filetype == "jsonl"}
         if backend == "cudf":
             read_f = cudf.read_json
+            if input_meta is not None:
+                read_kwargs["prune_columns"] = True
```

Comment:
Do you think we should move away from `input_meta` in favor of a keyword like `dtype` (like Pandas' and cuDF's `read_json`) and having the user configure `prune_columns` themselves?

I think this would align with https://github.com/NVIDIA/NeMo-Curator/issues/50 too.

Reply:
I'm generally in favor of overhauling the IO helpers in the current setup for something better. When we tackle #50. I'll share more thoughts there, but moving to encouraging users using the read_xyz api's is easier. 
We can then have a common helper that based on the filetype directs to the relevant `read_xyz` api rather than the other way around where `read_json` goes to a common read method that handles different formats.

Regarding: `prune_columns` specifically: This change is important in newer versions of rapids because many public datasets like rpv1 do not have consistent metadata across all their files. If we do not prune columns to just ID & Text, cuDF will now fail with inconsistent metadata errors.